src/containers/BookmarkList.js

```diff
@@ -0,0 +1,159 @@
+import More from '../components/More'
+import React from 'react'
+import styled from 'styled-components'
+import styles from './BookmarkList.scss'
+import twreporterRedux from 'twreporter-redux'
+import { Bookmark, Confirmation } from 'twreporter-react-components'
+import { CSSTransitionGroup } from 'react-transition-group'
+import { connect } from 'react-redux'
+import { setHeaderInfo } from '../actions/header'
+import { AUTHOR_PAGE, LIGHT } from '../constants/index'
+
+// lodash
+import get from 'lodash/get'
+import findIndex from 'lodash/findIndex'
+
+const _ = {
+  get,
+  findIndex
+}
+
+const DEFAULT_LIMIT = 6
+const SORT = 'created_at'
+const DIALOG_CONTENT = '您確定要刪除這篇文章書籤？'
+const DIALOG_CONFIRM = '確定'
+const DIALOG_CANCEL = '取消'
+
+const { actions } = twreporterRedux
+const { getBookmarks, createBookmark, deleteBookmark } = actions
+
+const MoreContainer = styled.div`
+  display: ${props => (props.hasMore ? 'inline' : 'none')}
+`
+
+class BookmarkList extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      showConfirmation: false
+    }
+    this.getNextPage = this._getNextPage.bind(this)
+    this.handleLoadMore = this._handleLoadMore.bind(this)
+    this.handleBookmarkIconOnClick = this._handleBookmarkIconOnClick.bind(this)
+    this.onCancel = this._onCancel.bind(this)
+    this.onConfirm = this._onConfirm.bind(this)
+    this.bookmarkData = []
+    this.total = 0
+    this.targetSlug = ''
+  }
+
+  componentWillMount() {
+    this.getNextPage()
+    const { setHeaderInfo } = this.props
+    setHeaderInfo({
+      pageTheme: LIGHT,
+      pageType: AUTHOR_PAGE
+    })
+  }
+
+  _getNextPage() {
+    const offset = this.bookmarkData.length
+    this.props.getBookmarks(offset, DEFAULT_LIMIT, SORT)
+      .then(({ data, total }) => {
+        this.total = total
+        if (this.bookmarkData.length <= total) {
+          data.forEach((bookmark) => {
+            this.bookmarkData.push(bookmark)
+          })
+          this.forceUpdate()
```

Comment:
move `this.bookmarkData` to `this.state.bookmarkData`?

In this component, changes of bookmarkData(add/delete) will affect how this component renders. Hence, bookmarkData should be the state of this component.

BTW, you could move total into bookmarkData state.